app/src/main/java/com/bnyro/contacts/util/LocalContactsRepository.kt

```diff
@@ -23,73 +25,81 @@ class LocalContactsRepository(context: Context) : ContactsRepository {
     }
 
     override suspend fun createContact(contact: ContactData) {
-        val localContact = LocalContact(
-            displayName = contact.displayName,
-            firstName = contact.firstName,
-            surName = contact.surName,
-            nickName = contact.nickName,
-            organization = contact.organization
-        )
-        val contactId = DatabaseHolder.Db.localContactsDao().insertContact(localContact)
-        val dataItems = listOf(
-            contact.numbers.toDataItem(contactId, DataCategory.NUMBER),
-            contact.emails.toDataItem(contactId, DataCategory.EMAIL),
-            contact.addresses.toDataItem(contactId, DataCategory.ADDRESS),
-            contact.events.toDataItem(contactId, DataCategory.EVENT),
-            contact.notes.toDataItem(contactId, DataCategory.NOTE),
-            contact.websites.toDataItem(contactId, DataCategory.WEBSITE),
-            contact.groups.map {
-                DbDataItem(
-                    contactId = contactId,
-                    category = DataCategory.GROUP.value,
-                    value = it.title,
-                    type = -1
-                )
-            }
-        ).flatten()
-        DatabaseHolder.Db.localContactsDao().insertData(*dataItems.toTypedArray())
-        contact.photo?.let { saveProfileImage(contactId, it) }
+        withContext(Dispatchers.IO) {
```

Comment:
Please move that into the line above, so that it's
```
override fun createContact(contact: ContactData) = withContext(Dispatchers.IO) {
```

Reply:
This doesn't work

Reply:
This line could return null. So the final return type becomes `Unit?`
``` kotlin
contact.photo?.let { saveProfileImage(contactId, it) }
```

Reply:
Doing this solved the issue
```kotlin
 contact.photo?.let { saveProfileImage(contactId, it) } ?: Unit
```

Reply:
Oh okay, then we better keep it as it is now. Sorry for the false hint.